pkg/ee/validation/machine/providers.go

```diff
@@ -459,3 +489,234 @@ func getProjectIDOrTenantID(configVarResolver *providerconfig.ConfigVarResolver,
+	if err != nil {
+		return nil, fmt.Errorf("failed to get hetzner raw config, error: %w", err)
+	}
+	configVarResolver := providerconfig.NewConfigVarResolver(ctx, userClient)
+
+	serverType, err := configVarResolver.GetConfigVarStringValue(rawConfig.ServerType)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get the value of \"serverType\" field, error: %w", err)
+	}
+	token, err := configVarResolver.GetConfigVarStringValueOrEnv(rawConfig.Token, envHZToken)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get the value of hetzner \"token\" field, error: %w", err)
+	}
+	hClient := hcloud.NewClient(hcloud.WithToken(token))
+
+	size, _, err := hClient.ServerType.Get(ctx, serverType)
+	if err != nil {
+		return nil, fmt.Errorf("failed to list hetzner sizes, error: %w", err)
+	}
+
+	cpuReq, err := resource.ParseQuantity(strconv.Itoa(size.Cores))
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse machine cpu request to quantity, error: %w", err)
+	}
+	memReq, err := resource.ParseQuantity(fmt.Sprintf("%fG", size.Memory))
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse machine memory request to quantity, error: %w", err)
+	}
+	// Todo: verify if gb or not/ number of disks or size
+	storageReq, err := resource.ParseQuantity(fmt.Sprintf("%dG", size.Disk))
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse machine storage request to quantity, error: %w", err)
+	}
+
+	return NewResourceDetails(cpuReq, memReq, storageReq), nil
+}
+
+func getNutanixResourceRequirements(ctx context.Context, userClient ctrlruntimeclient.Client, config *types.Config) (*ResourceDetails, error) {
+	rawConfig, err := nutanixtypes.GetConfig(*config)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get nutanix raw config, error: %w", err)
+	}
+
+	// TODO: check if total cores or total cpus
+	var coresPerCPU int64
+	if rawConfig.CPUCores != nil {
+		coresPerCPU = *rawConfig.CPUCores
+	}
+	totalCPUs := rawConfig.CPUs
+	totalCPUCores := totalCPUs * coresPerCPU
+
+	cpuReq, err := resource.ParseQuantity(strconv.FormatInt(totalCPUCores, 10))
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse machine cpu request to quantity, error: %w", err)
+	}
+	memReq, err := resource.ParseQuantity(fmt.Sprintf("%dM", rawConfig.MemoryMB))
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse machine memory request to quantity, error: %w", err)
+	}
+	storageReq, err := resource.ParseQuantity(fmt.Sprintf("%dG", rawConfig.DiskSize))
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse machine storage request to quantity, error: %w", err)
+	}
+
+	return NewResourceDetails(cpuReq, memReq, storageReq), nil
+}
+
+func getDigitalOceanResourceRequirements(ctx context.Context, userClient ctrlruntimeclient.Client, config *types.Config) (*ResourceDetails, error) {
+	rawConfig, err := digitaloceantypes.GetConfig(*config)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get digitalOcean raw config, error: %w", err)
```

Comment:
NIT to be consistend with other errors below

```suggestion
		return nil, fmt.Errorf("failed to get DigitalOcean raw config, error: %w", err)
```